docs/openapi.yaml

```diff
@@ -0,0 +1,1143 @@
+      tags:
+        - Robots
+      parameters:
+        - in: path
+          name: robot_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 21
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Map"
+          description: Request successful
+        "404":
+          description: Not found - No robot with id exists
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+
+  /robots/{robot_id}/schedule:
+    get:
+      operationId: getRobotSchedule
+      summary: Lookup the schedule for the robot
+      description: |
+        ### Overview
+        Get the current schedule of a robot. The schedule is a list of time entries where the robot is scheduled to perform a certain mission.
+
+        The minimum start time and maximum end time for the schedule entries can be specified in the query. If none is provided, the default start time is current time and default end time is start time + one week.
+      tags:
+        - Robots
+      parameters:
+        - in: path
+          name: robot_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 21
+        - in: query
+          name: startTime
+          schema:
+            type: string
+            format: date-time
+            description: Minimum start time for scheduled missions
+          required: false
+        - in: query
+          name: endTime
+          schema:
+            type: string
+            format: date-time
+            description: Maximum end time for scheduled missions
+          required: false
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Schedule"
+          description: Request successful
+        "404":
+          description: Not found - No robot with id exists
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+
+  /robots/{robot_id}/event:
+    post:
+      operationId: postRobotEvent
+      summary: Create new event for robot
+      description: |
+        ### Overview
+        Adds a new event to the robot schedule
+
+        New entries to the schedule can be added as long as they don't conflict with already scheduled events.
```

Comment:
What would a conflict look like? Is it if they are scheduled for the same time? Should this by default add to mission queue?

Reply:
Conflict is that the new event overlaps in time with already scheduled events. I think the user then should be informed about the conflict and which event(s) it is in conflict with instead of some default action.

Reply:
Agreed, that this should be the behaviour once some time for the inspection has been requested by the user, but should the default behaviour be that it is just entered into a queue and the user would then specify a specific time window if that is desired? I would guess that for many missions one does not care that much about when it is performed, just that it happens within a reasonable time